agcaballero/walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,87 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
 
 public final class FindMeetingQuery {
+   
+  /** the number of minutes in a day */
+  private static final int MINUTES_IN_DAY = 24 * 60;
+  
+  /**
+   * Takes the events of the day and information about a potential meeting 
+   * and returns the time ranges in which this meeting could be scheduled
+   *
+   * @return a collection of TimeRanges in which the meeting could be scheduled
+   * @param events the collection of events scheduled for that day
+   * @param request the meeting request to be fulfilled (will have duration & attendees)
+   */ 
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> attendees = request.getAttendees();
+    ArrayList<TimeRange> times = new ArrayList<TimeRange>();
+     // minutes of day is + 1 to account for first and last minute of day
+    boolean[] minutes = new boolean[MINUTES_IN_DAY + 1];
+
+    for(int i = 0; i < minutes.length; i++) {
+      minutes[i] = true;
+    }
+
+    for(Event event: events) {
+      // if there's an overlap in attendees block off those times
+      if(attendeeOverlap(event.getAttendees(), attendees)) {
+        TimeRange range = event.getWhen();
+
+        for(int i = range.start(); i < range.end(); i++)
+          minutes[i] = false;
+      }
+    }
+
+    int start = 0; 
+    boolean available = minutes[start];
+
+    // add available times to times array
+    for(int i = 0; i < minutes.length; i++) {
+      // if this is part of an available time range
+      if(available) {
+        // then, if current minute is false or you've reached end of day, add a new time range
+        if(! minutes[i]  || i == MINUTES_IN_DAY) {
```

Comment:
is `i == MINUTES_IN_DAY` equivalent to `i == minutes.length - 1`? If so, the latter would be preferable, as it enforces consistency (if in the future minutes.length changes, this condition wouldn't break).

Reply:
I didn't realize the latter would be better, but they are the same thing, and I agree that would make it more flexible and consistent. I'll change it.

Reply:
Done